File: api-tester/src/main/java/com/revenuecat/apitester/java/PackageAPI.java
package com.revenuecat.apitester.java;

import androidx.annotation.Nullable;

import com.revenuecat.purchases.Package;
import com.revenuecat.purchases.PackageType;
import com.revenuecat.purchases.PresentedOfferingContext;
import com.revenuecat.purchases.models.StoreProduct;

import java.net.URL;

@SuppressWarnings({"unused"})
final class PackageAPI {
    static void check(final Package p) {
        final String identifier = p.getIdentifier();
        final PackageType packageType = p.getPackageType();
        final StoreProduct product = p.getProduct();
        final String offering = p.getOffering();
        final PresentedOfferingContext offeringContext = p.getPresentedOfferingContext();
        final @Nullable URL webCheckoutURL = p.getWebCheckoutURL();
    }

    static void check(final PackageType type) {
        switch (type) {
            case UNKNOWN:
            case CUSTOM:
            case LIFETIME:
            case ANNUAL:
            case SIX_MONTH:
            case THREE_MONTH:
            case TWO_MONTH:
            case MONTHLY:
            case WEEKLY:
        }
    }
}

File: api-tester/src/main/java/com/revenuecat/apitester/kotlin/PackageAPI.kt
package com.revenuecat.apitester.kotlin

import com.revenuecat.purchases.Package
import com.revenuecat.purchases.PackageType
import com.revenuecat.purchases.models.StoreProduct

@Suppress("unused", "UNUSED_VARIABLE")
private class PackageAPI {
    fun check(p: Package) {
        with(p) {
            val identifier: String = identifier
            val packageType: PackageType = packageType
            val product: StoreProduct = product
            val offering: String = offering
        }
    }

    fun check(type: PackageType) {
        when (type) {
            PackageType.UNKNOWN,
            PackageType.CUSTOM,
            PackageType.LIFETIME,
            PackageType.ANNUAL,
            PackageType.SIX_MONTH,
            PackageType.THREE_MONTH,
            PackageType.TWO_MONTH,
            PackageType.MONTHLY,
            PackageType.WEEKLY,
            -> {}
        }.exhaustive
    }
}
